cargo-dist/src/tasks.rs

```diff
@@ -1103,6 +1108,134 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
+            let variant = self.variant(variant_idx);
+            let target = &variant.target;
+            if target.contains("windows") || target.contains("linux-gnu") {
+                continue;
+            }
+            // Compute the artifact zip this variant *would* make *if* it were built
+            // FIXME: this is a kind of hacky workaround for the fact that we don't have a good
+            // way to add artifacts to the graph and then say "ok but don't build it".
+            let (artifact, binaries) =
+                self.make_executable_zip_for_variant(to_release, variant_idx);
+            target_triples.insert(target.clone());
+            let fragment = ExecutableZipFragment {
+                id: artifact.id,
+                target_triples: artifact.target_triples,
+                zip_style: artifact.archive.as_ref().unwrap().zip_style,
+                binaries: binaries
+                    .into_iter()
+                    .map(|(_, dest_path)| dest_path.file_name().unwrap().to_owned())
+                    .collect(),
+            };
+
+            if target == X64_MACOS {
+                x86_64 = Some(fragment.clone());
+            }
+            if target == ARM64_MACOS {
+                arm64 = Some(fragment.clone());
+            }
+
+            if do_rosetta_fallback && target == X64_MACOS {
+                // Copy the info but respecify it to be arm64 macos
+                let mut arm_fragment = fragment.clone();
+                arm_fragment.target_triples = vec![ARM64_MACOS.to_owned()];
+                artifacts.push(arm_fragment.clone());
+                arm64 = Some(arm_fragment);
+            }
+            artifacts.push(fragment);
+        }
+        if artifacts.is_empty() {
+            warn!("skipping Homebrew installer: not building any supported platforms (use --artifacts=global)");
+            return;
+        };
+
+        let release = self.release(to_release);
+        let app_name = release.app_name.clone();
+        let app_desc = release.app_desc.clone();
+        let app_license = release.app_license.clone();
+        let app_homepage_url = release.app_homepage_url.clone();
+
+        let formula_name = to_class_case(&app_name);
+
+        let installer_artifact = Artifact {
+            id: artifact_name,
+            target_triples: target_triples.into_iter().collect(),
+            archive: None,
+            file_path: artifact_path.clone(),
+            required_binaries: FastMap::new(),
+            checksum: None,
+            kind: ArtifactKind::Installer(InstallerImpl::Homebrew(HomebrewInstallerInfo {
+                arm64,
+                x86_64,
+                name: app_name,
+                formula_class: formula_name,
+                desc: app_desc,
+                license: app_license,
+                homepage: app_homepage_url,
+                // For now, assume all packages contain the same binaries
+                binaries: artifacts[0].binaries.clone(),
```

Comment:
This doesn't feel right; the formula currently assumes the binary list for ARM and x86_64 will always be the same.

Reply:
Is that a fundamental constraint of homebrew formulas, or just a lot nastier to implement?

(tbh this isn't a big deal, there's no way to make the targets diverge in binaries rn)

Reply:
I was trying to avoid having a potentially unnecessary `if Hardware::CPU` check, but I think I'll just add an `if/else` to the template that avoids it if both sets are the same.